marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/step/saveStep.sjs

```diff
@@ -0,0 +1,60 @@
+/**
+ Copyright (c) 2020 MarkLogic Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+'use strict';
+
+const Artifacts = require('/data-hub/5/artifacts/core.sjs');
+const Step = require("/data-hub/5/impl/step.sjs");
+const consts = require("/data-hub/5/impl/consts.sjs");
+
+var stepDefinitionType;
+var stepProperties;
+
+stepProperties = stepProperties.toObject();
+const stepName = stepProperties.name;
+
+xdmp.trace(consts.TRACE_STEP, `Saving step with name ${stepName} and type ${stepDefinitionType}`);
+
+let existingStep = fn.head(cts.search(cts.andQuery([
+  cts.collectionQuery("http://marklogic.com/data-hub/steps"),
+  cts.jsonPropertyValueQuery("stepDefinitionType", stepDefinitionType, "case-insensitive"),
+  cts.jsonPropertyValueQuery("name", stepName)
+])));
+
+if (existingStep) {
+  xdmp.trace(consts.TRACE_STEP, `Step with name ${stepName} and type ${stepDefinitionType} already exists, so will update`);
+  let updatedStep = Object.assign(existingStep.toObject(), stepProperties);
+  Artifacts.setArtifact(stepDefinitionType, stepName, updatedStep);
+} else {
+  xdmp.trace(consts.TRACE_STEP, `Step with name ${stepName} and type ${stepDefinitionType}  does not exist, so will create`);
+
+  // For now, can assume the stepDefinitionName based on the type. Can add stepDefinitionType as a parameter once we need
+  // more flexibility.
+  const stepDefinitionName = "mapping" === stepDefinitionType.toLowerCase() ? "entity-services-mapping" : "default-ingestion";
+
+  stepProperties.stepDefinitionName = stepDefinitionName;
+  stepProperties.stepDefinitionType = stepDefinitionType;
+  stepProperties.stepId = stepName + "-" + stepDefinitionType;
+
+  const stepDef = new Step().getStepByNameAndType(stepDefinitionName, stepDefinitionType);
+  const stepDefOptions = stepDef.options;
```

Comment:
The default collections for ingestion step should be the step name (current behavior) and for mapping step, it's the step name and the entity name(explorer requires that ).

Reply:
That logic still exists in the loadData/mapping artifact libraries. 

Should we not copy the default collections from the step definition? In which case - shouldn't we remove the default collections from the step definition? Do they serve any other purpose?

Pinging @ryanjdew  and @bsrikan  for insight.

Reply:
I think I see what to do - the default collections are currently applied in defaultArtifactSettings, which isn't called for step creation. I'll copy/paste that logic into defaultArtifact for mapping.sjs, and then I'll modify this to not pull the default collections from the step definition.

Reply:
Thanks @srinathgit - I updated the tests (and added manageIngestionStep.sjs) so the assertions are on the step name and also the entity type for mapping tests.